session/sshforward/sshprovider/agentprovider.go

```diff
@@ -10,51 +10,57 @@ import (
 	"time"
 
 	"github.com/moby/buildkit/session"
-	"github.com/moby/buildkit/session/sshforward"
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
-	"golang.org/x/sync/errgroup"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/metadata"
 )
 
 // AgentConfig is the config for a single exposed SSH agent
 type AgentConfig struct {
 	ID    string
 	Paths []string
+	Raw   bool
 }
 
-// NewSSHAgentProvider creates a session provider that allows access to ssh agent
-func NewSSHAgentProvider(confs []AgentConfig) (session.Attachable, error) {
-	m := map[string]source{}
-	for _, conf := range confs {
-		if len(conf.Paths) == 0 || len(conf.Paths) == 1 && conf.Paths[0] == "" {
-			conf.Paths = []string{os.Getenv("SSH_AUTH_SOCK")}
-		}
+func (conf AgentConfig) toRaw() (rawConfig, error) {
```

Comment:
maybe `toDialer()` would be better here, and returning `dialer/sshDialer` struct.

Reply:
Makes since, updated.